src/state/ingress_per_unit.py

```diff
@@ -87,4 +92,7 @@ def from_provider(
                     raise IngressPerUnitIntegrationDataValidationError(
                         "Validation of ingress per unit relation data failed."
                     ) from exc
+                except ValueError as exc:
+                    logger.error(str(exc))
+                    raise IngressPerUnitIntegrationDataValidationError(str(exc)) from exc
```

Comment:
I think `from exc` should already include the exception message and `logger.error` should be used to provide more context and we should remove it if we don't have much to add.